src/ads/InterstitialAd.ts

```diff
@@ -153,4 +141,27 @@ export class InterstitialAd extends MobileAd implements MobileAdInterface {
 
     return this._googleMobileAds.native.interstitialShow(this._requestId, options);
   }
+
+  addAdEventsListener<T extends AdEventType>(listener: AdEventsListener<T>) {
+    if (!isFunction(listener)) {
+      throw new Error("InterstitialAd.addAdEventsListener(*) 'listener' expected a function.");
+    }
+
+    return this._addAdEventsListener(listener);
+  }
+
+  addAdEventListener<T extends AdEventType>(type: T, listener: AdEventListener<T>) {
+    if (!isOneOf(type, Object.values(AdEventType))) {
+      throw new Error("InterstitialAd.addAdEventListener(*) 'type' expected an AdEventType value.");
+    }
+    if (!isFunction(listener)) {
+      throw new Error("InterstitialAd.addAdEventListener(_, *) 'listener' expected a function.");
+    }
+
+    return this._addAdEventListener(type, listener);
+  }
+
+  removeAllListeners(): void {
+    this._removeAllListeners();
+  }
```

Comment:
addAdEventsListener/AddAdEventListener/removeAllListeners here and in AppOpenAdd are exactly the same except for the messages having the type that threw the error. :thinking: 

Maybe the base class can have a _name property, and a constructor that sets the name, which we call with super the derived classes
Then you can implement all three methods concretely in the superclass, using the name from construction in the error messages, and the duplication goes away here